org.eclipse.draw2d/src/org/eclipse/draw2d/graph/CompoundHorizontalPlacement.java

```diff
@@ -34,6 +35,13 @@ class LeftRight {
 
 		@Override
 		public boolean equals(Object obj) {
+			if (this == obj) {
```

Comment:
LeftRight is package-private. I'd probably convert it to a record. Then equals method is generated automatically.

Reply:
good idea, changed that.